### PR TITLE
NPE if pending uploads exists on account removal

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Steps that triggered NPE:
- have pending uploads
- remove last existing account

-> NPE in getCurrentAndPendingUploadsForCurrentAccount() and other due to null in
AccountUtils.getCurrentOwnCloudAccount(mContext)

I changed the function to 
public static @Nullable Account getCurrentOwnCloudAccount(Context context)

now the IDE warns if one access a potentially nullable variable. Very useful!

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
👍

File: src/main/java/com/owncloud/android/datamodel/UploadsStorageManager.java [hunk 1]
```diff
@@ -374,18 +374,21 @@ private OCUpload createOCUploadFromCursor(Cursor c) {
...
+                            "==" + UploadResult.DELAYED_FOR_WIFI.getValue() +
+                            " OR " + ProviderTableMeta.UPLOADS_LAST_RESULT +
+                            "==" + UploadResult.LOCK_FAILED.getValue() +
+                            " OR " + ProviderTableMeta.UPLOADS_LAST_RESULT +
+                            "==" + UploadResult.DELAYED_FOR_CHARGING.getValue() +
+                            " AND " + ProviderTableMeta.UPLOADS_ACCOUNT_NAME + "== ?",
+                    new String[]{account.name}
+            );
+        } else {
+            return new OCUpload[0];
```
Comment (mario) [inline src/main/java/com/owncloud/android/datamodel/UploadsStorageManager.java]: What if OCUpload[0] doesn't exist? :P

Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/datamodel/UploadsStorageManager.java]: This creates an OCUpload array with size 0.

In java repl:
java> String[] test = new String[0];
java.lang.String[] test = []
java> test.length
java.lang.Integer res2 = 0

Comment (mario) [reply src/main/java/com/owncloud/android/datamodel/UploadsStorageManager.java]: Geez. I haven't seen "new". Sorry.

**Comment by mario** [comment]
👍

**Pull request merged**